drizzlepac/hlautils/svm_quality_analysis.py

```diff
@@ -132,7 +140,134 @@ def compare_num_sources(catalog_list, drizzle_list, log_level=logutil.logging.NO
+
+    Parameters
+    ----------
+    hap_obj : drizzlepac.hlautils.Product.FilterProduct
+        hap filter product object to process
+
+    log_level : int, optional
+        The desired level of verboseness in the log statements displayed on the screen and written to the .log file.
+        Default value is 'NOTSET'.
+
+    Returns
+    --------
+    nothing.
+    """
+    log.setLevel(log_level)
+    slNames = [hap_obj.point_cat_filename,hap_obj.segment_cat_filename]
+    imgNames = [hap_obj.drizzle_filename, hap_obj.drizzle_filename]
+    good_flag_sum = 255 # all bits good
+
+    diag_obj = du.HapDiagnostic(log_level=log_level)
+    diag_obj.instantiate_from_hap_obj(hap_obj,
+                                      data_source="{}.compare_ra_dec_crossmatches".format(__taskname__),
+                                      description="matched point and segment catalog RA and Dec values")
+    json_results_dict = collections.OrderedDict()
+    # add reference and comparision catalog filenames as header elements
+    json_results_dict["point catalog filename"] = slNames[0]
+    json_results_dict["segment catalog filename"] = slNames[1]
+
+    # 1: Read in sourcelists files into astropy table or 2-d array so that individual columns from each sourcelist can be easily accessed later in the code.
+    point_data, seg_data = csl.slFiles2dataTables(slNames)
+    log.info("Valid point data columns:   {}".format(list(point_data.keys())))
+    log.info("Valid segment data columns: {}".format(list(seg_data.keys())))
+    log.info("\n")
+    log.info("Data columns to be compared:")
+    columns_to_compare = list(set(point_data.keys()).intersection(set(seg_data.keys())))
+    for listItem in sorted(columns_to_compare):
+        log.info(listItem)
+    log.info("\n")
+    # 2: Run starmatch_hist to get list of matched sources common to both input sourcelists
+    slLengths = [len(point_data['RA']), len(seg_data['RA'])]
+    json_results_dict['point catalog length'] = slLengths[0]
+    json_results_dict['segment catalog length'] = slLengths[1]
+    matching_lines_ref, matching_lines_img = csl.getMatchedLists(slNames, imgNames, slLengths, log_level=log_level)
+    json_results_dict['number of cross-matches'] = len(matching_lines_ref)
+    if len(matching_lines_ref) == 0 or len(matching_lines_img) == 0:
+        log.critical("*** Comparisons cannot be computed. No matching sources were found. ***")
+        return ("ERROR")
+    # 2: Create masks to remove missing values or values not considered "good" according to user-specified good bit values
+    # 2a: create mask that identifies lines any value from any column is missing
+    missing_mask = csl.mask_missing_values(point_data, seg_data, matching_lines_ref, matching_lines_img, columns_to_compare)
+    # 2b: create mask based on flag values
+    matched_values = csl.extractMatchedLines("FLAGS", point_data, seg_data, matching_lines_ref, matching_lines_img)
```

Comment:
The column name in the *ecsv file is "Flags".  There is no checking for case in the code, so this line will not work as expected.

Reply:
This is true, but it's not a problem. I wrote compare_sourcelists.py to handle different types of sourcelists (.ecsv hap sourcelists, and .sexphot.txt and daophot.txt HLA sourcelists) which have different names for the same columns (like "Flags" vs. "FLAGS). Compare_sourcelists.py has code that determines the sourcelist type, and maps the columns in the sourcelist to a set of in-program column names that are always the same, regardless of sourcelist type, so it's always an apples to apples comparison.
"FLAGS" as seen on line 206 is the in-program column name. All flag data point_data and seg_data is stored under this name, so it's actually working the way it should.

Reply:
ahhh.